hashring/hashring.go

```diff
@@ -187,81 +181,69 @@ func (r *HashRing) AddRemoveServers(add []string, remove []string) bool {
 	return changed
 }
 
-// hasServer returns true if the server exists in the ring, false otherwise
+// HasServer returns true if the server exists in the ring, false otherwise
 func (r *HashRing) HasServer(address string) bool {
 	r.servers.RLock()
-	server := r.servers.byAddress[address]
-	r.servers.RUnlock()
-
-	return server
+	defer r.servers.RUnlock()
+	return r.servers.byAddress[address]
 }
 
-func (r *HashRing) GetServers() (servers []string) {
+// GetServers returns all servers stored in the HashRing
+func (r *HashRing) GetServers() []string {
 	r.servers.RLock()
+	defer r.servers.RUnlock()
+
+	var servers []string
 	for server := range r.servers.byAddress {
 		servers = append(servers, server)
 	}
-	r.servers.RUnlock()
-
 	return servers
 }
 
 // ServerCount returns the number of servers in the ring
 func (r *HashRing) ServerCount() int {
 	r.servers.RLock()
-	count := len(r.servers.byAddress)
-	r.servers.RUnlock()
-
-	return count
+	defer r.servers.RUnlock()
+	return len(r.servers.byAddress)
 }
 
+// Lookup return the owner of the given key
 func (r *HashRing) Lookup(key string) (string, bool) {
-	r.servers.RLock()
-
-	iter := r.servers.tree.IterAt(int(r.hashfunc([]byte(key))))
-	if iter.Nil() {
-		r.servers.RUnlock()
+	strs := r.LookupN(key, 1)
+	if len(strs) == 0 {
 		return "", false
```

Comment:
should we return nil instead of "" while we're at it?  Would that be valid here?